src/main/java/com/demo/classroom/DTO/ApiResponse.java

```diff
@@ -0,0 +1,24 @@
+package com.demo.classroom.DTO;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiResponse<T> {
+    private boolean success;
+    private String message;
+    private T data; 
+    
+
+    public ApiResponse(boolean success, String message) {
+        this(success, message, null);
+    }
+
+    public ApiResponse(boolean success, String message, T data) {
+        this.success = success;
+        this.message = message;
```

Comment:
If message is a required field for every response, consider validating or defaulting it in the constructors.
```suggestion
        this.message = message != null ? message : "";

```

Reply:
Solved